horace_core/Tobyfit/tobyfit_DGdisk_resconv_init.m

```diff
@@ -259,14 +259,53 @@
         sample_coords_to_spec_to_rlu(wtmp.experiment_info);
     
     % Get detector information
-    % Because detpar only contains minimal information, hardwire in the detector type here
+    % Because detpar only contains minimal information, either hardwire in 
+    % the detector type here or use the info now available in the detector
+    % arrays
     detpar = wtmp.detpar();   % just get a pointer
-    if use_tubes
-        detectors{iw} = IX_detector_array (detpar.group, detpar.x2(:), detpar.phi(:), detpar.azim(:),...
-            IX_det_He3tube (detpar.width, detpar.height, 6.35e-4, 10));   % 10atms, wall thickness=0.635mm
+    det = wtmp.experiment_info.detector_arrays;
+    if isempty(det) || det.n_runs == 0
+        % no detector info was available when the sqw was populated, so
+        % continue with the old detector initialisation from detpar
+        if use_tubes
+            detectors{iw} = IX_detector_array (detpar.group, detpar.x2(:), detpar.phi(:), detpar.azim(:),...
+                IX_det_He3tube (detpar.width, detpar.height, 6.35e-4, 10));   % 10atms, wall thickness=0.635mm
```

Comment:
Thanks for the comment, but why these particular numbers? MAPS instrument?

Reply:
@oerc0122 these numbers are in the original code and in various other tests in the codebase. I am assuming they will be representative of such tubes, and will be overridden by actual data when it is available in nxspe